src/utils/rfq_blockchain_utils.ts

```diff
@@ -114,6 +115,35 @@ export class RfqBlockchainUtils {
         return this._exchangeProxy.executeMetaTransaction(metaTx, metaTxSig).getABIEncodedTransactionData();
     }
 
+    public async getNonceAsync(workerAddress: string): Promise<number> {
+        return this._web3Wrapper.getAccountNonceAsync(workerAddress);
+    }
+
+    public getExchangeProxyAddress(): string {
+        return this._exchangeProxyAddress;
+    }
+
+    public async getTransactionReceiptIfExistsAsync(transactionHash: string): Promise<TransactionReceipt | undefined> {
+        return this._web3Wrapper.getTransactionReceiptIfExistsAsync(transactionHash);
+    }
+
+    public async getCurrentBlockAsync(): Promise<number> {
+        return this._web3Wrapper.getBlockNumberAsync();
+    }
+
+    public async estimateGasForExchangeProxyCallAsync(callData: string, workerAddress: string): Promise<number> {
+        const txData: Partial<TxData> = {
+            to: this._exchangeProxy.address,
+            data: callData,
+            from: workerAddress,
+        };
+
+        const gasEstimate = await this._web3Wrapper.estimateGasAsync(txData);
```

Comment:
Why do we estimate every single time? can we hard-code an estimate instead? we should know upfront what the upper and lower bounds for this trade is, right? can we hard-code and avoid an RPC call?

Reply:
it can change based on on-chain conditions, but that's a bit paranoid--we can just rely on the buffer

Reply:
If we hard-coded something like 180k, we should be safe right? that's like a 10-15% buffer.

Reply:
probably not, some tokens can cost much higher